src/sql/src/plan/query.rs

```diff
@@ -2094,6 +2107,119 @@ fn plan_table_factor(
     )
 }
 
+// The following SQL is equivalent:
+//
+// SELECT
+//     gs1.generate_series, expand.x, expand.n, gs2.generate_series
+// FROM
+//     (SELECT row_number() OVER (), * FROM ROWS FROM (generate_series(1, 2))) AS gs1
+//     FULL JOIN (SELECT row_number() OVER (), * FROM ROWS FROM (information_schema._pg_expandarray(ARRAY[9])))
+//             AS expand ON gs1.row_number = expand.row_number
+//     FULL JOIN (SELECT row_number() OVER (), * FROM ROWS FROM (generate_series(3, 6))) AS gs2 ON
+//             gs1.row_number = gs2.row_number;
+//
+// and:
+//
+// SELECT
+//     *
+// FROM
+//     ROWS FROM (
+//         generate_series(1, 2),
+//         information_schema._pg_expandarray(ARRAY[9]),
+//         generate_series(3, 6)
+//     );
+//
+// This function creates a HirRelationExpr that is identical to what is
+// produced by the former.
+fn plan_rows_from(
+    ecx: &ExprContext,
+    functions: &[TableFunction<Aug>],
+    alias: Option<&TableAlias>,
+) -> Result<(HirRelationExpr, Scope), anyhow::Error> {
+    let mut funcs = Vec::with_capacity(functions.len());
+    let mut project_outputs = Vec::new();
+    let mut last = 0;
```

Comment:
A note here as to what `last` does would be helpful, e.g. projects away the row number column, which we'll reorder to be first in the the first column in each return set.